lib/Dialect/RTL/Ops.cpp

```diff
@@ -522,8 +549,23 @@ void XorOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
         return success();
       }
 
+      if (inputs[size - 1] == inputs[size - 2]) {
+        if (size == 2) {
+          // xor(x, x) -> xor ( 0 ) -- idempotent
+          auto zero = rewriter.create<ConstantOp>(
+              op.getLoc(), 0, op.getType().cast<IntegerType>());
+
+          SmallVector<Value, 1> zeroOperand({zero});
+          rewriter.replaceOpWithNewOp<XorOp>(op, op.getType(), zeroOperand);
```

Comment:
Actually looking at this code, it is probably better to handle the xor(x,x) case in the `XorOp::fold` method.  This is checked before the canonicalization patterns and is used more widely.

The trick there is you'll have to form the zero as an attribute.  Using something like this should work:

if (size == 2 && they are the same)
  return IntegerAttr::get(getType(), 0);